counterpartylib/lib/util.py

```diff
@@ -198,6 +202,82 @@ def get_asset_name (db, asset_id, block_index):
     elif not assets:
         return 0    # Strange, I know…
 
+# checks and validates subassets (PARENT.SUBASSET)
+#   throws exceptions for assset or subasset names with invalid syntax
+#   returns (None, None) if the asset is not a subasset name
+def parse_subasset_from_asset_name(asset):
+    subasset_parent = None
+    subasset_child = None
+    subasset_longname = None
+    chunks = asset.split('.', 1)
+    if (len(chunks) == 2):
+        subasset_parent = chunks[0]
+        subasset_child = chunks[1]
+        subasset_longname = asset
+
+        # validate parent asset
+        validate_subasset_parent_name(subasset_parent)
+
+        # validate child asset
+        validate_subasset_longname(subasset_longname, subasset_child)
+
+    return (subasset_parent, subasset_longname)
+
+# takes an asset description in 
+#   if it has a suffix like ;;lPARENT.child, then parse the subasset information
+#   throws exceptions for invalid asset names
+def parse_subasset_from_description(description):
+    subasset_parent = None
+    subasset_longname = None
+    chunks = description.rsplit(';;l', 1)
+    if len(chunks) == 2:
+        split_description, asset = chunks
+        subasset_parent, subasset_longname = parse_subasset_from_asset_name(asset)
+        if subasset_longname is not None:
+            # if we parsed a subasset, strip ;;l{subasset} from the description
+            description = split_description
+
+    return (subasset_parent, subasset_longname, description)
+
+# throws exceptions for invalid subasset names
+def validate_subasset_longname(subasset_longname, subasset_child):
+    if len(subasset_child) < 1:
+        raise exceptions.AssetNameError('subasset name too short')
+    if len(subasset_longname) > 250:
```

Comment:
I think a 40 character limit is the most practical, but since the CIP was discussed and a bounty raised, I think we should leave it.  My opinion is not strong.  I think the anti-spam fee is big enough to prevent harmful abuse and wallet makers can just ignore long assets.